### ISSUE TITLE
Caching JSON

### ISSUE THREAD

**Issue by taylortrimble** [teslaworksumn/teslaworks.net#18]
It's been bugging me that we parse JSON every time we load a page in this app. Instead, we can load it with the `before_first_request` decorator and then reload it every time we update the data.

Alternatively, check if files have been updated, then parse if that's the case. We can do this with the `before_request` decorator.

**Comment by mplewis** [comment]
Ideally, make some pages static and rebuild on demand :D

**Comment by taylortrimble** [comment]
I'd rather not have half loaded in a static way and half loaded in a dynamic way.

Our landing would have to be dynamic (Basecamp query for most recent activity). Our project ones could by static... for now. we lose the architecture that lets us do future dynamic things in the project pages (BC integrations maybe).

I'm not super excited about static pages, especially since we'd have to serve them in a dynamic way anyway:

``` python
@app.route(/<project>)
def project():
    if 'project' in projects:
        if 'redirect_url' in keys:
            redirect_to_blog_link() # we would want the actual blog link, not a copy
        else:
            serve_some_static_page_from_inside_a_dynamic_function()
    else:
        abort(404)
```

Plus, we still need the templates for generating static content. And sing the templates dynamically is not slow.

**Comment by mplewis** [comment]
In any case, the static suggestion is kind of tongue-in-cheek. Because why
optimize before we have to? :D

**Comment by mplewis** [comment]
Also also you forget the magic of client-side JSON re: loading stuff from a
server.

**Comment by mplewis** [comment]
I meant JavaScript. This post is getting awkward.

**Comment by taylortrimble** [comment]
I thought about that, but it doesn’t seem to fit with the rest of Flask. Plus Python interface are a _lot_ nicer than the equivalent JavaScript ones!

Especially since the only thing that would change would be `load_static_thing()` to `use_this_template()` and the workflow goes from:
1. Update data model
2. Write out JSON (notice what I forgot earlier: no reload required, we updated our model)

To:
1. Update data model
2. Write out JSON
3. Regenerate static stuff

> This post is getting awkward.

Lol, J in JSON stands for JavaScript. Yeah. :smile:

**Comment by mplewis** [comment]
Counter-point: almost every site you use uses client-side JavaScript in
some way to provide dynamic content. Including Basecamp. (:

However, I like keeping it simple for now! Like you said.

**Comment by taylortrimble** [comment]
I knew that you were going to say that. But since the templating is used for the home page we might as well use it everywhere so there’s “one way” – and you’re officially okay with that?

**Comment by mplewis** [comment]
Yes!

_stop using my buzzwords against me_

**Comment by mplewis** [comment]
Has this been closed with #42?

**Comment by taylortrimble** [comment]
Yeppers!
